app/src/services/cpk.ts

```diff
@@ -1569,6 +1576,19 @@ class CPKService {
       throw e
     }
   }
+
+  claimAirdrop = async ({ account, setTxHash, setTxState }: CPKClaimAirdropParams) => {
+    const network = await this.provider.getNetwork()
+    const networkId = network.chainId
+    const signer = this.provider.getSigner()
+    const address = await signer.getAddress()
+    const txOptions: TxOptions = {}
+    await this.getGas(txOptions)
+
+    const airdrop = new AirdropService(networkId, this.provider, address, this.cpk.relay)
+    const transactions: Transaction[] = await airdrop.encodeClaims(account)
+    return this.execTransactions(transactions, txOptions, setTxHash, setTxState)
+  }
 }
```

Comment:
Assuming any address can call the claim function for any other address, should we have the user perform this transaction directly instead of using the CPK, to save gas?

Reply:
I think it was originally setup to allow claims directly but because we enabled multiple airdrops to run simultaneously a user may need to send a tx through the cpk to claim from multiple airdrops.

🤔 could just check if there are multiple tho and do direct if not

Reply:
Yeah, in that case I think this is a good implementation. I'm more concerned about preventing bugs which might occur than reducing gas costs for an edge case.